### DISCUSSION TITLE
discussion: Found some code quality issues in edgedb

### DISCUSSION THREAD

**Discussion by pnijhara** [geldata/gel#1528] [Show and tell]
I have recently forked this repo and went to a code walk through. There I found some code quality issues. Later I ran a [DeepSource](https://deepsource.io) analysis on it.

DeepSource found more than 8000 code quality issues which are categorized as:

- **Antipatterns:** 4.3k out of which 800+ can be autofixed in a single click
- **Bug-risks:** 3.5k
- **Performance issues: 195**
- **Security issues: 600** out of which around 500 can be autofixed.

You can see the detailed report of the analysis here -> https://deepsource.io/gh/pnijhara/edgedb/issues/?category=recommended

I have also added a deepsource configuration file for letting DeepSource analyze the repo. You can find the configuration file [here](https://github.com/pnijhara/edgedb/blob/pnijhara-patch-1/.deepsource.toml)

I can send a patch of some fixes if you want. Also, do let me know which of them do you consider as false positives and which of them as a potential issue.

**Comment by 1st1** [comment]
I looked through the report. There are indeed some nits here and there, but no bugs, serious code style problems, or most importantly security issues (all are false positives). The performance analysis of DeepSource is very shallow, fixing the issues it found wouldn't make any detectable improvements. 

Summing this up: we are open for PRs, for sure, but please remove our org from your marketing lists.

**Reply by pnijhara** [reply]
I will send a PR with some fixes.
